package datto_linux_client;

import "base_signature.proto";

message StartBackupRequest {
    enum BackupType {
        FULL_BACKUP = 1;
        INCREMENTAL_BACKUP = 2;
        DIFF_BACKUP = 3;
    }

    required string block_path = 1;
    required BackupType type = 2;

    required string destination_host = 3;
    // There is no uint16 so use this
    required uint32 destination_port = 4;
    
    optional BaseSignature base_signature = 5;
}
